pkg/entities/message.go

```diff
@@ -20,6 +20,16 @@ const (
 	MaxTcpSocketMsgSize uint16 = 65535
 )
 
+// data struct of processed message
+type Message struct {
+	Version      uint16
+	BufferLength uint16
+	SeqNumber    uint32
+	ObsDomainID  uint32
+	ExportTime   uint32
+	Record       interface{}
```

Comment:
Feel Sets is the right name as there can be multiple set inside a message.

Reply:
Yes I'm considering merge MsgBuffer and Message together so that we can have uniform interface for exporter and collector, then Sets can be used here.

Reply:
Sounds good.

Reply:
yes, maybe in the PR to move encoding to io stream or a separate PR.